### PR TITLE
Spr6 followup & fix some errors

### PR DESCRIPTION
**Pull request by ernesto-glz**
## Changes Proposed:
-  Finished the implementation of spr6.
- Fix some compatibility bugs with new versions of nest and ORM
- prevent crash when authguard is executed and a token is not passed with headers or cookies

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: src/auth/account.repository.ts [hunk 1]
```diff
@@ -20,6 +20,18 @@ import { Misc } from '../shared/misc';
 
 @EntityRepository(Account)
 export class AccountRepository extends Repository<Account> {
+  private accountInformationRepo = getRepository(
+    AccountInformation,
+    'authConnection',
+  );
+
+  private accountPasswordRepo = getRepository(
```
Comment (IntelligentQuantum) [inline src/auth/account.repository.ts:28]: Not a good way, but good for now

**Review by IntelligentQuantum — APPROVED** [review]

File: src/auth/account.repository.ts (hunk 1 above)
Comment (ernesto-glz) [reply src/auth/account.repository.ts:28]: what would be the best way to do this?, in the documentation I did not find anything about it

Comment (IntelligentQuantum) [reply src/auth/account.repository.ts:28]: @ernesto-glz we have connection here https://github.com/azerothcore/acore-api/blob/master/src/auth/auth.module.ts#L11

Strange that gives an error.
You have to make a repository file

Comment (ernesto-glz) [reply src/auth/account.repository.ts:28]: @IntelligentQuantum  It only gives an error when I am trying to use some method directly from an entity like here https://github.com/azerothcore/acore-api/blob/master/src/auth/account.repository.ts#L36

**Pull request merged**